consoles/ssh_screen.pm

```diff
@@ -52,4 +59,41 @@ sub do_read
     }
     return undef;
 }
+
+sub type_string
+{
+    my ($self, $nargs) = @_;
+
+    bmwqemu::log_call(%$nargs);
+
+    my $text           = $nargs->{text};
+    my $terminate_with = $nargs->{terminate_with} // '';
+    my $written        = 0;
+
+    $text .= "\cC" if ($terminate_with eq 'ETX');
+
+    while ($written < length($text)) {
+        my $chunk = $self->ssh_channel->write(substr($text, $written));
+
+        if (!defined($chunk)) {
+            my ($errcode, $errname, $errstr) = $self->ssh_connection->error;
+
+            croak "Lost SSH connection to SUT: $errstr"
+              if $errcode != LIBSSH2_ERROR_EAGAIN;
+            select(undef, undef, undef, 0.1);
+        } elsif ($chunk < 0) {
+            # Old Net::SSH2 error signaling
+            croak "Lost SSH connection to SUT"
+              if $chunk != LIBSSH2_ERROR_EAGAIN;
+            select(undef, undef, undef, 0.1);
+        } else {
+            $written += $chunk;
+        }
+    }
+
+    if ($terminate_with eq 'EOT') {
+        $self->ssh_channel->send_eof;
```

Comment:
I suggest to reduce some whitespace here, e.g.


```perl
$self->ssh_channel->send_eof if $terminate eq 'EOT';
```

Reply:
Fixed.